### ISSUE TITLE
Specify the output layer name to do the prediction

### ISSUE THREAD

**Issue by Avsecz** [kipoi/kipoi#222]
Ideally, the api would look like this:

```bash
kipoi predict <model> ... --layer=<mylayer> ...
```
If layer is none, then the final layer would be used. One could also specify a dictionary or a list of layers (corresponding to the list or dictionary of returned arrays).

- Give a few useful brakepoints in the description of the model
  - layer before logits etc...

## TODO

- [ ] add the BcolzWriter to kipoi

## Open question

- [x] does pytorch have names for the layers to do something like this?

**Comment by Avsecz** [comment]
@krrome is it possible to get predictions for a specific layer which we can refer by name?

**Comment by Avsecz** [comment]
I guess it would be easy to do if the pytorch model is `torch.nn.Sequential`

**Comment by krrome** [comment]
That's easy for Keras, but not quite so for pytorch. Pytorch offers a way to assign unique names to layers when using sequentials and more general models. I think the output could be captured with forward hooks, I still have to take a look into that and will update this comment when I have understood more. I have to do similar things for gradient calculation btw. so you can leave that to me.

**Comment by Avsecz** [comment]
Let me know if you need any help

**Comment by Avsecz** [comment]
closed by https://github.com/kipoi/kipoi/pull/237

**Comment by Avsecz** [comment]
I'll add the bcolz writer as a separate issue
